src/components/schedule-page/view-only-page/view-only-toolbar.tsx

```diff
@@ -1,26 +1,81 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { ImportButtonsComponent } from "../import-buttons/import-buttons.component";
 import { Link } from "react-router-dom";
 import { Button } from "../../common-components";
+import { useDispatch, useSelector } from "react-redux";
+import { ApplicationStateModel } from "../../../state/models/application-state.model";
+import { RevisionReducerActionCreator } from "../../../state/reducers/month-state/revision-info.reducer";
+import { isRevisionType, RevisionTypeLabels } from "../../../api/persistance-store.model";
+import classNames from "classnames/bind";
+import { VerboseDateHelper } from "../../../helpers/verbose-date.helper";
 
 interface ViewOnlyToolbarOptions {
   openEdit: () => void;
 }
 export function ViewOnlyToolbar({ openEdit }: ViewOnlyToolbarOptions): JSX.Element {
+  const [isRevisionEditDisabled, setIsRevisionEditDisable] = React.useState<boolean>(false);
+  const [isMonthFromFuture, setIsMonthFromFuture] = React.useState<boolean>(false);
+  const dispatch = useDispatch();
+
+  const { year, month_number: month } = useSelector(
+    (state: ApplicationStateModel) => state.actualState.persistentSchedule.present.schedule_info
+  );
+
+  const { revision } = useSelector((state: ApplicationStateModel) => state.actualState);
+
+  useEffect(() => {
+    const isFuture = VerboseDateHelper.isMonthInFuture(month, year);
+    setIsMonthFromFuture(isFuture);
+    if (revision === "actual") {
+      setIsRevisionEditDisable(false);
+    } else {
+      setIsRevisionEditDisable(!isFuture);
+    }
+  }, [year, month, revision]);
+
+  const handleChange = (event: React.ChangeEvent<{ name?: string; value: string }>): void => {
+    const currentRev = event.target.value;
+    if (isRevisionType(currentRev)) {
+      dispatch(RevisionReducerActionCreator.changeRevision(currentRev));
+    }
+  };
+
   return (
     <div className="buttons">
+      <div className="revision-type-container">
+        {isMonthFromFuture ? (
+          <p>{RevisionTypeLabels[revision]}</p>
+        ) : (
+          <form>
+            <select
+              value={revision}
+              onChange={handleChange}
+              className="revision-select"
+              data-cy="revision-select"
+            >
+              <option value="primary" data-cy="primary-revision">
+                {RevisionTypeLabels["primary"]}
+              </option>
+              <option value="actual" data-cy="actual-revision">
+                {RevisionTypeLabels["actual"]}
+              </option>
+            </select>
+          </form>
+        )}
+      </div>
       <div className="filler" />
       <ImportButtonsComponent />
       <Link to="/schedule-editing">
         <Button
           onClick={openEdit}
           size="small"
-          className="submit-button"
+          className={classNames(isRevisionEditDisabled && "disabled-submit-button")}
```

Comment:
warunek się jakoś tak robiło: 
["disabled-submit-button"]: isRevisionEditDisabled

Reply:
A faktycznie, zmienię to :)

Reply:
'''
classNames({"disabled-submit-button": isRevisionEditDisabled})
""
Działa w ten sposób